doc/releases/changelog-0.21.0.md

```diff
@@ -41,27 +150,15 @@
    DeviceArray([ 0.26043, -0.26043], dtype=float32, weak_type=True))
   ```
   Note that `jax.jit` is not supported for vector-valued QNodes.
-  [(#2110)](https://github.com/PennyLaneAI/pennylane/pull/2110)
 
-* For subclasses of `Operator` where it is known before instantiation, the `num_params` is reverted back to being a
-  static property. This allows to programmatically know the number of parameters before an operator is
-  instantiated without changing the user interface. A test was added to ensure that
-  different ways of defining `num_params` work as expected.
-  [(#2099)](https://github.com/PennyLaneAI/pennylane/pull/2099)
-  [(#2101)](https://github.com/PennyLaneAI/pennylane/pull/2101)
-  [(#2135)](https://github.com/PennyLaneAI/pennylane/pull/2135)
-
-* Development of circuit cutting compiler has begun:
-  A `WireCut` operator has been added for manual wire cut placement
-  when constructing a QNode.
-  [(#2093)](https://github.com/PennyLaneAI/pennylane/pull/2093)
+<h4>Generalized RotosolveOptmizer</h4>
 
 * The `RotosolveOptimizer` has been generalized to arbitrary frequency spectra
   in the cost function. Also note the changes in behaviour listed under *Breaking
   changes*.
   [(#2081)](https://github.com/PennyLaneAI/pennylane/pull/2081)
 
-  Previously, the `RotsolveOptimizer` was available for cost functions with
+  Previously, the `RotosolveOptimizer` was available for cost functions with
   frequency spectra that only contained integers, and the maximal frequency
   (instead of the number of frequencies) determined the cost of the optimization.
   Now arbitrary frequencies are supported.
```

Comment:
I think this will need to be re-written, because on reading it now, I realize it won't be clear to a casual reader _what_ the improvement actually is (that is, what can you do with Rotosolve that you couldn't do before?).

@antalszava I'm curious, did you have trouble understanding the impact from the changelog entry?

I wonder if we can rewrite this to remove mentions of frequencies in the first paragraph (fine to mention it further down), but instead emphasize what you can now do with Rotosolve that you couldn't do before. E.g.,

- "Previously, the `RotosolveOptimizer` was available for cost functions with frequency spectra that only contained integers" --> "Previously, the `RotosolveOptimizer` only supported variational circuits with single-qubit Pauli rotations".

- "Now arbitrary frequencies are supported." --> "Now, circuits with arbitrary gates are supported natively without decomposition, as long as the frequencies of the gate parameters are known."

- "the maximal frequency (instead of the number of frequencies) determined the cost of the optimization." --> "This new generalization extends the Rotosolve optimization method to a larger class of circuits, and can reduce the cost of the optimization compared to decomposing all gates to single-qubit rotations."

Reply:
@dwierichs, tagging you here in case you have any thoughts/in case I have said anything wrong above :)

Reply:
> @antalszava I'm curious, did you have trouble understanding the impact from the changelog entry?

I believe so, yes :smile: Frankly, I would have had to dive a bit deeper in there to grasp the concrete change. I recalled the restriction with `RotosolveOptimizer`, the suggestions make it clear from a usability point of view.

> Previously, the RotosolveOptimizer only supported variational circuits with single-qubit Pauli rotations

I do wonder about this. If I recall correctly, the constraint was as follows (from the Rotosolve paper):

![Screenshot from 2022-02-04 11-36-36](https://user-images.githubusercontent.com/24476053/152567422-8b456a78-41aa-4c10-9150-1e76f893fc97.png)
 
As such, single-qubit Pauli rotations satisfy this requirement, but may not be the only ones that do so (not truly sure if there are other practical cases really in the qubit model. :sweat_smile:). So I'm wondering if we'd like to update it to "Previously, the RotosolveOptimizer only supported variational circuits using special gates such as single-qubit Pauli rotations".

Applying these suggestions and can update them as we'd like. :)

Reply:
You are right, I was oversimplifying it!

Reply:
My five cents:

> As such, single-qubit Pauli rotations satisfy this requirement, but may not be the only ones that do so

Indeed, any (i.e also multi-qubit) Pauli rotations are included in the original Rotosolve scope, because multi-qubit Pauli words also only have two distinct eigenvalues.
I suppose the question here also is to which version we are comparing? The original Rotosolve had the above restriction, the intermediate generalization also allowed other gates as long as the frequencies were integer-valued (including gates like `CRX` or `SingleExcitation` etc), and the full generalization released in 0.21 also includes more fancy gates like QAOA cost layers of non-integer-valued Hamiltonians and such.

The rest of the discussion here looks spot-on to me :)